android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -524,7 +526,27 @@ private void setAudioFocus(boolean focus) {
              * speaker mode if this is not set.
              */
             audioManager.setMode(AudioManager.MODE_IN_COMMUNICATION);
-            audioManager.setSpeakerphoneOn(!audioManager.isWiredHeadsetOn());
+            audioManager.setSpeakerphoneOn(!(audioManager.isBluetoothScoOn() || audioManager.isBluetoothA2dpOn()));
```

Comment:
I notice that for wired headsets you have the potential to turn the speakerphone on and then off very quickly.  Would it make more sense to have the loop below set a variable like `hasNonSpeakerphoneDevice` and then just setting the speaker phone at the end based on that flag?